src/trrntzip.c

```diff
@@ -160,10 +160,8 @@ WORKSPACE *AllocateWorkspace(void) {
 }
 
 void FreeWorkspace(WORKSPACE *ws) {
-  if (ws->FileNameArray)
-    DynamicStringArrayDestroy(ws->FileNameArray, ws->iElements);
-  if (ws->pszCheckBuf != NULL)
-    free(ws->pszCheckBuf);
+  DynamicStringArrayDestroy(ws->FileNameArray, ws->iElements);
```

Comment:
Why is this safe? AFAICT, `DynamicStringArrayDestroy` doesn't check its first argument for `!= NULL`

Reply:
Oops, I thought that I'd set `iElements` to zero when the allocation failed, and then `DynamicStringArrayDestroy()` doesn't dereference the pointer. But apparently I forgot that.
So should I clear `iElements`, add back the `if`, or both?

Reply:
I think putting back the `if` is the simplest change, but if you want to do both, go for it :)

Reply:
Should still be safe because `FreeWorkspace()` is never called when the initial `FileNameArray` allocation fails.